src/DDTrace/Integrations/Nette/NetteIntegration.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace DDTrace\Integrations\Nette;
+
+use DDTrace\Contracts\Scope;
+use DDTrace\GlobalTracer;
+use DDTrace\Integrations\Integration;
+use DDTrace\SpanData;
+use DDTrace\Tag;
+use DDTrace\Type;
+use Nette\Application\Application;
+
+class NetteIntegration extends Integration
+{
+    const NAME = 'nette';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return self::NOT_AVAILABLE;
+        }
+
+        if (\PHP_MAJOR_VERSION < 7) {
+            $integration = $this;
+            \DDTrace\hook_method('Nette\Configurator', '__construct', function () use ($integration) {
+                $integration->load();
+            });
+        } else {
+            $this->load();
+        }
+
+        return self::LOADED;
+    }
+
+    public function load()
+    {
+        $scope = GlobalTracer::get()->getRootScope();
+        if (!$scope instanceof Scope) {
+            return;
+        }
+
+        $service = \ddtrace_config_app_name(NetteIntegration::NAME);
+
+        $root = $scope->getSpan();
+        $root->setTraceAnalyticsCandidate();
+        $root->overwriteOperationName('nette.request');
+        $root->setTag(Tag::SERVICE_NAME, $service);
+
+        \DDTrace\trace_method(
+            'Nette\Configurator',
+            'createContainer',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.configurator.createContainer';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Configurator',
+            'createRobotLoader',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.configurator.createRobotLoader';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
```

Comment:
Is there any other meta-data that can be attached to `createRobotLoader` or `createContainer`? Based on their names, they seem like the kind of thing we should trace only if there is additional meta-data.

Reply:
As far as I can tell there's really no useful data associated with them, they were just ported over from the original PR.

Reply:
@SammyK I think we should drop these ones; what do you think?

Reply:
@morrisonlevi My apologies; I discussed this with @Anilm3 off thread and forgot to update this thread with the discussion:

> As a general rule, I think it is still useful to trace functions even if there is no metadata if that function has the possibility to call something slow. For example, some sort of plugin loader; maybe on most requests it’s a small span with no metadata, but if there’s a slow plugin doing something strange on intermittent requests, it’s nice to know it’s happening from the plugin loader. I don’t know what `createRobotLoader` does, but if it loads possibly slow application logic, it might be worth tracing still.

It looks like the [Robot Loader](https://github.com/nette/robot-loader) does some crawling+compiling of PHP files and caches the classes/paths. So perhaps it would be nice to see the non-cached timing of that? It can also [throw an exception](https://github.com/nette/bootstrap/blob/8df6751/src/Bootstrap/Configurator.php#L200-L218) so the exception would have a span to attach to. 

For `createContainer`, I'm not sure if [container initialization has potential to be slow](https://github.com/nette/bootstrap/blob/8df6751/src/Bootstrap/Configurator.php#L231-L243), but if it did, I would think it beneficial to instrument.

But I don't have enough experience with Nette to say one way or another, so I'll leave the final call and PR approval up to you @morrisonlevi.

Reply:
> As a general rule, I think it is still useful to trace functions even if there is no metadata if that function has the possibility to call something slow.

I believe that in general this should be a concern for the profiler. That doesn't have to be absolute, but I think the exceptions should be for major framework concepts that are prominent in the docs. Are these things critical and major pieces to Nette applications? My gut says no, but I haven't looked at the Nette docs much.

Also, @Anilm3, message me on Slack because I think we should look at a Nette app with the profiler on -- when I did that with Laravel it was really illuminating (ha, an unintentional pun).

Reply:
I took a look in the [v3 docs](https://api.nette.org/3.0/Nette.html) and the robot loader does seem worth instrumenting. Ideally we'd try to find some meta-data for it, but that kind of thing can be added later.

I'm not sure about createContainer. It seems like tracing something in Container could be useful; maybe remove this for now?

Reply:
I've removed `createContainer`, thanks for double checking.